zerver/models/groups.py

```diff
@@ -82,6 +82,32 @@ class Meta:
         unique_together = (("realm", "name"),)
 
 
+class NamedUserGroup(UserGroup):  # type: ignore[django-manager-missing] # django-stubs cannot resolve the custom CTEManager yet https://github.com/typeddjango/django-stubs/issues/1023
+    MAX_NAME_LENGTH = 100
+
+    usergroup_ptr = models.OneToOneField(
+        auto_created=True,
+        on_delete=CASCADE,
+        parent_link=True,
+        primary_key=True,
+        serialize=False,
+        to=UserGroup,
+        related_name="named_user_group",
+    )
```

Comment:
Probably worth a comment on why we need to spell this out.

Reply:
Added comments to explain adding the `usergroup_ptr` field as well as the `related_name` field.